File: Java/LowLevel/src/com/eduardoprogramador/lowlevel/LowLevel.java
/*
 *   Copyright 2022. Eduardo Programador
 *   www.eduardoprogramador.com
 *   consultoria@eduardoprogramador.com
 *
 *   All Rights Reserved.
 *
 * */

package com.eduardoprogramador.lowlevel;

public class LowLevel {

    public static String reversed(String chars) {
        String reversed = "";
        for (int i = chars.length() - 1; i >-1 ; i--) {
            if(chars.charAt(i) == '0') {
                reversed += 0;
            } else {
                reversed += 1;
            }
        }
        return reversed;
    }

    public static class NumberSystem {

        private NumberSystem() {}

        public static NumberSystem init() {
            return new NumberSystem();
        }

        public String decimalToBinary(String decimal) {
            String res = "";
            long number = Long.valueOf(decimal);
            while ((number / 2) >= 1) {
                res += number % 2;
                number /= 2;
            }
            res += number;
            String resF = "";
            for (int i = res.length() - 1; i > -1 ; i--) {
                resF += res.charAt(i);
            }
            return resF;
        }
    }
}

File: Python/LowLevel/LowLevel.py
#/**
# *   Copyright 2022. Eduardo Programador
# *   www.eduardoprogramador.com
# *   consultoria@eduardoprogramador.com
# *
# *   All Rights Reserved.
# *
# * */

class LowLevel:

    @staticmethod
    def reversed(chars):
        res = ""
        i = len(chars) - 1
        while i >-1:
            if(chars[i] == '0'):
                res += "0"
            else:
                res += "1"
            i -= 1
        return res

    class NumberSystem:

        @classmethod
        def init(cls):
            return cls()

        @classmethod
        def decimalToBinary(cls,decimal):
            res = ""
            number = int(decimal)
            while int(number / 2) >= 1:
                res += str(number % 2)
                number = int(number / 2)
            res += str(number)
            resF = ""
            i = len(res) - 1
            while i > -1:
                resF += str(res[i])
                i -= 1
            return resF